src/events/events.ts

```diff
@@ -0,0 +1,391 @@
+ * @param before the member's profile state before update
+ * @param after the member's current profile state
+ */
+export const onMemberUpdate = async (client: Client, before: GuildMember, after: GuildMember): Promise<void> => {
+    // The server and logs channel ID of the server that the member updated their profile in
+    const server: Server = await Server.get(before.guild.id);
+    const logsID: string = server.logs;
+
+    // Skip if the server isn't subscribed to logs
+    if (!logsID) {
+        return;
+    }
+
+    const logs: TextChannel = <TextChannel> client.channels.cache.get(logsID);
+    const embed: EmbedBuilder = genMemberUpdateEmbed(before, after);
+
+    // If embed is null, the changes aren't significant to Sushi Bot
+    if (!embed) {
+        return;
+    }
+
+    try {
+        // Log to the logs channel
+        await logs.send({ embeds: [embed] });
+    } catch {
+        // Remove configurations if logging failed.
+        // Most likely causes are that the channel was deleted or that
+        // Sushi Bot had its permissions revoked
+        server.logs = null;
+        await server.save();
+    }
+};
+
+/**
+ * Logs user profile update events to servers subscribed to logs.
+ * 
+ * @param client the Discord bot
+ * @param before the user's profile state before update
+ * @param after the user's current profile state
+ */
+export const onUserUpdate = async (client: Client, before: User, after: User): Promise<void> => {
+    const embed: EmbedBuilder = genUserUpdateEmbed(before, after);
+
+    // If embed is null, the changes aren't significant to Sushi Bot
+    if (!embed) {
+        return;
+    }
+
+    // All mutual servers
+    const guilds: Collection<string, Guild> = client.guilds.cache;
+
+    // For each server...
+    for await (const [, guild] of guilds) {
+        try {
+            // If fetched, the user is a member of that server
+            await guild.members.fetch(before.id);
+        } catch {
+            // If not, they're not, so we skip
+            continue;
+        }
+
+        // The server and logs channel ID of the mutual server
+        const server: Server = await Server.get(guild.id);
+        const logsID: string = server.logs;
+
+        // Skip if the server isn't subscribed to logs
+        if (!logsID) {
+            continue;
+        }
+
+        const logs: TextChannel = <TextChannel> client.channels.cache.get(logsID);
+
+        try {
+            // Log to the logs channel
+            await logs.send({ embeds: [
+                embed.setAuthor({
+                    name: "Profile updated",
+                    iconURL: guild.iconURL()
+                })
+            ]});
+        } catch {
+            // Remove configurations if logging failed.
+            // Most likely causes are that the channel was deleted or that
+            // Sushi Bot had its permissions revoked
+            server.logs = null;
+            await server.save();
+        }
+    }
+};
```

Comment:
This listener gets triggered when a user makes changes to their global profile; and broadcasts the changes, if they are significant to Sushi Bot's logs, to all servers' logs channels